tools/mason/MasonArgParse.chpl

```diff
@@ -186,46 +202,99 @@ module MasonArgParse {
       // make sure all options defined got values if needed
       _checkSatisfiedOptions();
 
+      // assign and missing values their defaults, if supplied
+      _assignDefaultsToMissingOpts();
+
       // check for when arguments passed but none defined
       if argsList.size > 0 && this._actions.size == 0 {
         throw new ArgumentError("unrecognized options/values encountered: " +
                                 " ".join(argsList.these()));
       }
     }
 
+    proc _assignDefaultsToMissingOpts() {
+      // set any default values as needed
+      for name in this._actions.keys() {
+        const act = this._actions.getBorrowed(name);
+        const arg = this._result.getReference(name);
+        if !arg._present && !act._defaultValue.isEmpty() {
+          arg._values.extend(act._defaultValue);
+          arg._present = true;
+        }
+      }
+    }
+
     proc _checkSatisfiedOptions() throws {
-      // make sure we satisfied options that need at least 1 value
+      // make sure we satisfied options that require a value
       for name in this._actions.keys() {
         const act = this._actions.getBorrowed(name);
         const arg = this._result.getReference(name);
-        if act._numArgs.low > 0 && !arg._present {
+        if act._required && !arg._present {
           throw new ArgumentError("\\".join(act._opts) + " not enough values");
         }        
       }
     }
 
-    // define a new string option with fixed number of values expected
+    // define a new string option with fixed number of values expected no
+    // default value
+    proc addOption(name:string,
+                   opts:[]string,
+                   numArgs:int,
+                   required=false) throws {      
+      return addOption(name=name,
+                      opts=opts,
+                      numArgs=numArgs..numArgs,
+                      required=required,
+                      defaultValue=new list(string));
+    }
+
+    proc addOption(name:string,
+                   opts:[]string,
+                   numArgs:int,
+                   required=false,
+                   defaultValue:list(string)) throws {      
+      return addOption(name=name,
+                      opts=opts,
+                      numArgs=numArgs..numArgs,
+                      required=required,
+                      defaultValue=defaultValue);
+    }
+    
+    // define a new string option with fixed number of values expected and a
+    // single default value
```

Comment:
Seems like it would be easier to maintain if you fold many of these `addOption` calls together into a generic one. You can use `defaultValue:?t = none` and allow `string` or `list(string)` as arguments.

Reply:
Yes! Thank you for suggesting this. I believe @dlongnecke-cray was pointing me in this direction previously. This will allow me to remove two of the overloads on `addOption`